src/steps/os/macos.rs

```diff
@@ -93,3 +94,104 @@ pub fn run_sparkle(ctx: &ExecutionContext) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn update_xcodes(ctx: &ExecutionContext) -> Result<()> {
+    let xcodes = require("xcodes")?;
+    print_separator("Xcodes");
+
+    let should_ask = !(ctx.config().yes(Step::Xcodes)) || ctx.config().dry_run();
+
+    // `xcodes update` will update the list and display the list as an output.
+    let releases = ctx
+        .run_type()
+        .execute(&xcodes)
+        .args(["update"])
+        .output_checked_utf8()?
+        .stdout;
+
+    let releases_installed: Vec<String> = releases
+        .lines()
+        .filter(|release| release.contains("(Installed)"))
+        .map(String::from)
+        .collect();
+
+    if releases_installed.is_empty() {
+        println!("No Xcode releases installed.");
+        return Ok(());
+    }
+
+    let (allow_gm, allow_beta, allow_regular) =
+        releases_installed
+            .iter()
+            .fold((false, false, false), |(gm, beta, regular), release| {
+                (
+                    gm || release.contains("GM"),
+                    beta || release.contains("Beta"),
+                    regular || !(release.contains("GM") || release.contains("Beta")),
+                )
+            });
+
+    let releases_filtered: Vec<String> = releases
+        .lines()
+        .filter(|release| {
+            (allow_gm && release.contains("GM"))
+                || (allow_beta && release.contains("Beta"))
+                || (allow_regular && !release.contains("GM") && !release.contains("Beta"))
+        })
+        .map(String::from)
+        .collect();
+
+    if !releases_filtered
+        .last()
```

Comment:
It seems that I am missing something? Why are we only doing this to the `.last()` one?

Reply:
The last one is the most recent release on the list. If it contains the "(Installed)" substring, then Xcode is fully up to date. On the contrary, it'll then ask if one would like to upgrade to the newest release.

Reply:
Assume `xcodes update` gives the following output:

```
1.0 GM (0) (Installed)
1.1 GM (1)
1.4 Beta (0) (Installed)
1.5 Beta (1)
2.2.1 (0) (Installed)
2.2.2 (1)
```
`allow_gm`, `allow_beta` and `allow_regular` will all be `true`, `releases_filtered` will contain all the listed versions, in this case, if I understand correctly, we should install `1.1 GM (1)`, `1.5 Beta (1)` and `2.2.2 (1)` all rather than only the last one?

Reply:
So update each release? I can do that, will commit that soon :^)

Reply:
> So update each release? 

Emm, I am sorry, I thought this is the original implementation plan :<

>  I can do that, will commit that soon :^)

Thanks for doing that!